prow/cmd/tide/main.go

```diff
@@ -159,33 +150,86 @@ func main() {
 	if err != nil {
 		logrus.WithError(err).Fatal("Error constructing mgr.")
 	}
-	c, err := tide.NewController(
-		githubSync,
-		githubStatus,
-		mgr,
-		cfg,
-		git.ClientFactoryFrom(gitClient),
-		o.maxRecordsPerPool,
-		opener,
-		o.historyURI,
-		o.statusURI,
-		nil,
-		o.github.AppPrivateKeyPath != "",
-	)
-	if err != nil {
-		logrus.WithError(err).Fatal("Error creating Tide controller.")
+
+	if cfg().Tide.Gerrit != nil && cfg().Tide.Queries.QueryMap() != nil && o.providerName == "" {
+		logrus.Fatal("Both github and gerrit are configured in tide config but provider is not set.")
 	}
+
+	var c *tide.Controller
+	var gitClient *prowgit.Client
+	if cfg().Tide.Gerrit != nil {
```

Comment:
I am confused, we now have the provider flag but we never use it to determine what controller to use? Or am I missing it somehow?

Reply:
fixed